programs/fileio.c

```diff
@@ -157,7 +159,60 @@ static void clearHandler(void)
 }
 
 
-/* ************************************************************
+/*-*********************************************************
+*  Termination signal trapping (Print debug stack trace)
+***********************************************************/
+#define MAX_STACK_FRAMES    50
+
+#ifndef _WIN32
+static void ABRThandler(int sig) {
+   const char* name;
+   void* addrlist[MAX_STACK_FRAMES + 1];
+   char** symbollist;
+   U32 addrlen, i;
+
+   switch (sig) {
+      case SIGABRT: name = "SIGABRT"; break;
+      case SIGFPE: name = "SIGFPE"; break;
+      case SIGILL: name = "SIGILL"; break;
+      case SIGINT: name = "SIGINT"; break;
+      case SIGSEGV: name = "SIGSEGV"; break;
+      default: name = "UNKNOWN";
+   }
+
+   DISPLAY("Caught %s signal, printing stack:\n", name);
+   // Retrieve current stack addresses.
+   addrlen = backtrace(addrlist, sizeof(addrlist) / sizeof(void*));
```

Comment:
minor : I would have selected `sizeof(addrlist) / sizeof(*addrlist)` for automatic type adaptation